data/json/items/gunmod/underbarrel.json

```diff
@@ -43,7 +43,15 @@
     "color": "light_red",
     "location": "underbarrel",
     "mod_targets": [ "rifle" ],
-    "gun_data": { "ammo": "shot", "skill": "shotgun", "range": 2, "dispersion": 345, "durability": 10, "clip_size": 2 },
+    "gun_data": {
+      "ammo": "shot",
+      "skill": "shotgun",
+      "range": 2,
+      "dispersion": 345,
+      "durability": 10,
+      "clip_size": 2,
+      "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "shot": 2 } } ]
```

Comment:
this won't do anything, for any of the gunmods as is

Reply:
None of these should do anything currently. Gunmods that define `clip_size` should end up working like anything else with an integral magazine so they need pockets.

Reply:
i think we should put off adding the json for gunmods until they're read so we can avoid the json reporting unvisited members.

Reply:
Moved pocket definitions out of `gun_data`.